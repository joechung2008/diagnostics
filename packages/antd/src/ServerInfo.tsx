import { Table } from "antd";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  const columns = [
    {
      title: "Name",
      dataIndex: "name",
      key: "name",
    },
    {
      title: "Value",
      dataIndex: "value",
      key: "value",
    },
  ];

  return (
    <Table
      aria-label="Server Info"
      columns={columns}
      dataSource={items.map((item) => ({ ...item, key: item.name }))}
      pagination={false}
    />
  );
};

export default ServerInfo;
