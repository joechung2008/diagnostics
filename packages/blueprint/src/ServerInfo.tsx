import { HTMLTable } from "@blueprintjs/core";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <HTMLTable aria-label="Server Info">
      <thead>
        <tr>
          <th>Name</th>
          <th>Value</th>
        </tr>
      </thead>
      <tbody>
        {items.map((item) => (
          <tr key={item.name}>
            <td>{item.name}</td>
            <td>{item.value}</td>
          </tr>
        ))}
      </tbody>
    </HTMLTable>
  );
};

export default ServerInfo;
