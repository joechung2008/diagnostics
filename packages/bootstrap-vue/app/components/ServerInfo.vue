<template>
  <div>
    <BTable :items="rows" :fields="fields" small />
  </div>
</template>

<script setup lang="ts">
import type { ServerInfoProps } from "@/types/ServerInfoProps";
import { when } from "@/utils/extensions";

const props = defineProps<ServerInfoProps>();

const fields = [
  { key: "name", label: "Name" },
  { key: "value", label: "Value" }
];

const rows = computed(() => [
  { name: "Hostname", value: props.hostname },
  ...when(props.uptime !== undefined, {
    name: "Uptime",
    value: String(props.uptime)
  }),
  { name: "Server ID", value: props.serverId },
  { name: "Deployment ID", value: props.deploymentId },
  ...when(props.nodeVersions !== undefined, {
    name: "Node Versions",
    value: props.nodeVersions
  }),
  {
    name: "Extension Sync | Total Sync All Count",
    value: String(props.extensionSync.totalSyncAllCount)
  }
]);
</script>
