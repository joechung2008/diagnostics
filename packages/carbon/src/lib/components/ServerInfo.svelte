<script lang="ts">
	import { DataTable } from 'carbon-components-svelte';

	const { deploymentId, extensionSync, hostname, nodeVersions, serverId, uptime }: ServerInfoProps =
		$props();

	const headers = [
		{ key: 'name', value: 'Name', empty: false },
		{ key: 'value', value: 'Value', empty: false }
	];

	const rows = $derived([
		{ id: 'hostname', name: 'Hostname', value: hostname },
		{ id: 'uptime', name: 'Uptime', value: uptime },
		{ id: 'server-id', name: 'Server ID', value: serverId },
		{ id: 'deployment-id', name: 'Deployment ID', value: deploymentId },
		{ id: 'node-versions', name: 'Node Versions', value: nodeVersions },
		{
			id: 'extension-sync',
			name: 'Extension Sync | Total Sync All Count',
			value: extensionSync.totalSyncAllCount
		}
	]);
</script>

<DataTable aria-label="Server Information" {headers} {rows} />
