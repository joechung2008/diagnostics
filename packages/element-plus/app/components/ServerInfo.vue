<template>
    <el-table :data="rows" style="width: 100%" aria-label="Server Info">
        <el-table-column prop="name" label="Name" show-overflow-tooltip />
        <el-table-column prop="value" label="Value" show-overflow-tooltip />
    </el-table>
</template>

<script setup lang="ts">
import type { ServerInfoProps } from "@/types"

const props = defineProps<ServerInfoProps>()

const rows = computed(() => [
    { name: "Hostname", value: props.hostname },
    { name: "Uptime", value: props.uptime },
    { name: "Server ID", value: props.serverId },
    { name: "Deployment ID", value: props.deploymentId },
    { name: "Node Versions", value: props.nodeVersions },
    {
        name: "Extension Sync | Total Sync All Count",
        value: props.extensionSync.totalSyncAllCount
    }
])
</script>
