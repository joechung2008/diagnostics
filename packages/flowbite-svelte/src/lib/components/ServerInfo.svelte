<script lang="ts">
  import {
    Table,
    TableBody,
    TableBodyCell,
    TableBodyRow,
    TableHead,
    TableHeadCell
  } from "flowbite-svelte";

  let { deploymentId, extensionSync, hostname, nodeVersions, serverId, uptime }: ServerInfoProps =
    $props();

  const items = $derived.by(() => [
    { name: "Hostname", value: hostname },
    { name: "Uptime", value: uptime },
    { name: "Server ID", value: serverId },
    { name: "Deployment ID", value: deploymentId },
    { name: "Node Versions", value: nodeVersions },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount
    }
  ]);
</script>

<Table aria-label="Server Info">
  <TableHead>
    <TableHeadCell>Name</TableHeadCell>
    <TableHeadCell>Value</TableHeadCell>
  </TableHead>
  <TableBody>
    {#each items as item (item.name)}
      <TableBodyRow>
        <TableBodyCell>{item.name}</TableBodyCell>
        <TableBodyCell>{item.value}</TableBodyCell>
      </TableBodyRow>
    {/each}
  </TableBody>
</Table>
