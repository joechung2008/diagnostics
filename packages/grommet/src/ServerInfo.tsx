import { Table, TableBody, TableCell, TableHeader, TableRow } from "grommet";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <Table aria-label="Server Info">
      <TableHeader>
        <TableRow>
          <TableCell scope="col">Name</TableCell>
          <TableCell scope="col">Value</TableCell>
        </TableRow>
      </TableHeader>
      <TableBody>
        {items.map((item) => (
          <TableRow key={item.name}>
            <TableCell>{item.name}</TableCell>
            <TableCell>{item.value}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
};

export default ServerInfo;
