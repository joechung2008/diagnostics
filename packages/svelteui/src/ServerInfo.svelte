<script lang="ts">
  export let deploymentId: string
  export let extensionSync: { totalSyncAllCount: number }
  export let hostname: string
  export let nodeVersions: string
  export let serverId: string
  export let uptime: number

  const items = [
    {
      name: 'Hostname',
      value: hostname
    },
    {
      name: 'Uptime',
      value: uptime
    },
    {
      name: 'Server ID',
      value: serverId
    },
    {
      name: 'Deployment ID',
      value: deploymentId
    },
    {
      name: 'Node Versions',
      value: nodeVersions
    },
    {
      name: 'Extension Sync | Total Sync All Count',
      value: extensionSync.totalSyncAllCount
    }
  ]
</script>

<div class="table-container">
  <table aria-label="Server Information">
    <thead>
      <tr>
        <th>Name</th>
        <th>Value</th>
      </tr>
    </thead>
    <tbody>
      {#each items as item (item.name)}
        <tr>
          <td>{item.name}</td>
          <td>{item.value}</td>
        </tr>
      {/each}
    </tbody>
  </table>
</div>

<style scoped>
  table {
    width: 100%;
    border-collapse: collapse;
    margin: 1rem 0;
  }

  table th,
  table td {
    padding: 0.75rem 1rem;
    text-align: left;
  }

  table th {
    font-weight: 600;
    font-size: 0.875rem;
    text-transform: uppercase;
    letter-spacing: 0.05em;
  }

  table td {
    font-size: 0.875rem;
  }

  table td:first-child {
    font-weight: 500;
  }

  .table-container {
    overflow-x: auto;
  }
</style>
