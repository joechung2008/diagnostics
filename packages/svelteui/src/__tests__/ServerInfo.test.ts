import { render } from '@testing-library/svelte'
import { describe, expect, it } from 'vitest'
import ServerInfo from '../ServerInfo.svelte'

describe('ServerInfo', () => {
  it('should render server information correctly', () => {
    const { container } = render(ServerInfo, {
      props: {
        deploymentId: 'deploy-12345',
        extensionSync: { totalSyncAllCount: 42 },
        hostname: 'web-server-01',
        nodeVersions: 'v18.17.0',
        serverId: 'srv-abc123',
        uptime: 3600
      }
    })

    expect(container).toMatchSnapshot()
  })

  it('should render with different values', () => {
    const { container } = render(ServerInfo, {
      props: {
        deploymentId: 'prod-deploy-v2.1',
        extensionSync: { totalSyncAllCount: 156 },
        hostname: 'api-gateway-02',
        nodeVersions: 'v20.5.1',
        serverId: 'srv-def456',
        uptime: 86400
      }
    })

    expect(container).toMatchSnapshot()
  })

  it('should render with zero uptime', () => {
    const { container } = render(ServerInfo, {
      props: {
        deploymentId: 'test-deploy',
        extensionSync: { totalSyncAllCount: 0 },
        hostname: 'localhost',
        nodeVersions: 'v19.8.1',
        serverId: 'srv-test001',
        uptime: 0
      }
    })

    expect(container).toMatchSnapshot()
  })
})
