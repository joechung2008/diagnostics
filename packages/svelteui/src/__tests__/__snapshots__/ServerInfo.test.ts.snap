// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerInfo > should render server information correctly 1`] = `
<div>
  <div
    class="table-container svelte-1dur3k8"
  >
    <table
      aria-label="Server Information"
      class="svelte-1dur3k8"
    >
      <thead>
        <tr>
          <th
            class="svelte-1dur3k8"
          >
            Name
          </th>
           
          <th
            class="svelte-1dur3k8"
          >
            Value
          </th>
        </tr>
      </thead>
       
      <tbody>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Hostname
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            web-server-01
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Uptime
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            3600
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Server ID
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            srv-abc123
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Deployment ID
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            deploy-12345
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Node Versions
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            v18.17.0
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Extension Sync | Total Sync All Count
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            42
          </td>
           
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`ServerInfo > should render with different values 1`] = `
<div>
  <div
    class="table-container svelte-1dur3k8"
  >
    <table
      aria-label="Server Information"
      class="svelte-1dur3k8"
    >
      <thead>
        <tr>
          <th
            class="svelte-1dur3k8"
          >
            Name
          </th>
           
          <th
            class="svelte-1dur3k8"
          >
            Value
          </th>
        </tr>
      </thead>
       
      <tbody>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Hostname
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            api-gateway-02
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Uptime
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            86400
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Server ID
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            srv-def456
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Deployment ID
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            prod-deploy-v2.1
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Node Versions
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            v20.5.1
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Extension Sync | Total Sync All Count
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            156
          </td>
           
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`ServerInfo > should render with zero uptime 1`] = `
<div>
  <div
    class="table-container svelte-1dur3k8"
  >
    <table
      aria-label="Server Information"
      class="svelte-1dur3k8"
    >
      <thead>
        <tr>
          <th
            class="svelte-1dur3k8"
          >
            Name
          </th>
           
          <th
            class="svelte-1dur3k8"
          >
            Value
          </th>
        </tr>
      </thead>
       
      <tbody>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Hostname
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            localhost
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Uptime
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            0
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Server ID
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            srv-test001
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Deployment ID
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            test-deploy
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Node Versions
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            v19.8.1
          </td>
           
        </tr>
        <tr>
          <td
            class="svelte-1dur3k8"
          >
            Extension Sync | Total Sync All Count
          </td>
           
          <td
            class="svelte-1dur3k8"
          >
            0
          </td>
           
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
